benchmarks/benchmark.ml

```diff
@@ -63,6 +110,14 @@ let exec_bench_many exec name re cases =
       cases |> List.iter ~f:(fun x -> ignore (exec re x))
     )
 
+let rec read_all_http pos re reqs =
+  if pos >= String.length reqs
+  then ()
+  else
+    let g = Re.exec ~pos re reqs in
+    let (_, pos) = Re.Group.offset g 0 in
+    read_all_http (pos + 1) re reqs
```

Comment:
It is, except that I don't allocate a list here. This is preferable for the benchmark.

I suppose I could use all_gen here, but I'd have to verify the overhead is minimal.